go-runtime/ftl/ftltest/leases.go

```diff
@@ -0,0 +1,77 @@
+package ftltest
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/TBD54566975/ftl/go-runtime/ftl"
+	"github.com/TBD54566975/ftl/internal/modulecontext"
+)
+
+// mockLeaseClient is a simple in-memory lease client for testing.
+//
+// It does not include any checks on module names, as it assumes that all leases are within the module being tested
+type mockLeaseClient struct {
```

Comment:
We have another implementation of this, can we consolidate (actual question, this might not be possible)?

Also can we call this a ["fake"](https://martinfowler.com/articles/mocksArentStubs.html) instead of a "mock"?

To quote Martin Fowler:

> Fake objects actually have working implementations, but usually take some shortcut which makes them not suitable for production (an [in memory database](https://martinfowler.com/bliki/InMemoryTestDatabase.html) is a good example).
> Mocks are objects pre-programmed with expectations which form a specification of the calls they are expected to receive.